### DISCUSSION TITLE
Feature request: Ability to disable capture modes

### DISCUSSION THREAD

**Discussion by Herbis** [gopro/labs#676] [Feature Requests]
### Motivation
Many people, including myself, have the need to only use one of the modes, usually it's of course - Video mode. 
Sometimes, one can accidentally switch modes without realizing it, and start the capture, only to later find out, they recorder all their footage as a time-lapse, or bunch of captured photos. 

For me, this also likely occurred due to turning on the camera, accidentally switching to time lapse, and even noticing it. Then 'update date and time' popup would trigger, and while I fill that in, I forget about the mode switch, and just click record. 
That's how I got useless timelapse footage of my basketball game. 

### Expected behavior
1. Preferred - QRControl setting that allows to disable all but one of the "Photo", "Video", "Time-lapse" modes.
2. Alternate - QRControl setting that disallows switching of capture mode (without the use of QRControl app), but still allows to adjust capture settings through GoPro. 

### Known workarounds
1. There is an archive mode that can be enabled with QRControl code:
https://gopro.github.io/labs/control/archive/
Downside - it prevents doing anything else with the camera, in terms of switching video presets, adjusting zoom etc. That's very inconvenient. 

3. Using QRControl scan the preset of required camera mode, and setup date and time. 
Downside - have to use the phone every time. Inconvenient. 
Since I need to setup the cameras quickly, and I'm in my jersey at that time, I don't have my phone with me.

**Requesting for GoPro camera models:** Hero 11, Hero 10, future Hero models

Here are some reddit and GoPro forum posts looking for the same features, referencing lost footage and frustration:
https://community.gopro.com/s/question/0D53b00008PTyGQCA1/how-can-i-block-off-photo-and-time-lapse?language=en_US
https://www.reddit.com/r/gopro/comments/mjpiz4/turn_off_options_for_photo_and_time_lapse_mode/
https://www.reddit.com/r/gopro/comments/11ka2r5/any_way_to_disable_or_remove_time_lapse_and_or/
https://community.gopro.com/s/question/0D53b00008Qa4RzCAJ/disable-timelapse-from-mode-menu?language=en_US

**Comment by dnewman-gpsw** [comment]
Labs can't remove preset and modes from the carousel, it does modify the UI at all.  However you can run a boot script keep you in your preferred mode.  If you switch modes, it will switch back.  You cancel with any QR Code.

![image](https://github.com/gopro/labs/assets/10856084/85fcc733-a272-4ef0-9160-6bcbc7b4f6ef)

For HERO10 is doesn't support some the above command, just use !MBOOT="!Lvmode"!SAVEvmode=mV!R  
This does works on all cameras, however it will post a "processing" message on later models.

**Reply by Herbis** [reply]
Thank you! This seems to be working well for both my Hero 10 and Hero 11.

**Reply by dnewman-gpsw** [reply]
This is not going to be a good overall solution as it is a bit too aggressive in setting video mode every second. You will need to cancel the script to do some UI operations.  

I have added a request for conditionals that test the current mode so that the UI behaves correctly.

**Reply by Herbis** [reply]
Thanks, improvements would be greatly appreciated!

**Comment by dnewman-gpsw** [comment]
moved to discussions, as **Issues** postings get closed when solved.
